integrations-and-supported-tools/plotting-libraries/notebooks/Neptune_Plotting_Support.ipynb

```diff
@@ -0,0 +1,657 @@
+    "```\n",
+    "\n",
+    "To find your API token and full project name:\n",
+    "\n",
+    "1. [Log in to Neptune](https://app.neptune.ai/).\n",
+    "1. In the bottom-left corner, expand your user menu and select **Get your API token**.\n",
+    "1. The workspace name is displayed in the top-left corner of the app. \n",
+    "\n",
+    "    To copy the project path, in the top-right corner, open the settings menu and select **Properties**.\n",
+    "\n",
+    "For more help, see [Setting Neptune credentials](https://docs.neptune.ai/setup/setting_credentials) in the Neptune docs."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "run = neptune.init_run(\n",
+    "    api_token=neptune.ANONYMOUS_API_TOKEN,\n",
+    "    project=\"common/plotting\",\n",
+    "    tags=[\"notebook\"],\n",
+    ")"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {
+    "tags": [
+     "comment",
+     "exclude"
+    ]
+   },
+   "source": [
+    "**To open the run in the Neptune web app, click the link that appeared in the cell output.**\n",
+    "\n",
+    "We'll use the `run` object we just created to log metadata. You'll see the metadata appear in the app."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {
+    "tags": [
+     "header"
+    ]
+   },
+   "source": [
+    "## Log Altair charts to Neptune"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Create a sample chart"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%pip install -U -q altair vega_datasets"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import altair as alt\n",
+    "from vega_datasets import data\n",
+    "\n",
+    "source = data.cars()\n",
+    "\n",
+    "brush = alt.selection_interval()\n",
+    "\n",
+    "points = (\n",
+    "    alt.Chart(source)\n",
+    "    .mark_point()\n",
+    "    .encode(\n",
+    "        x=\"Horsepower:Q\",\n",
+    "        y=\"Miles_per_Gallon:Q\",\n",
+    "        color=alt.condition(brush, \"Origin:N\", alt.value(\"lightgray\")),\n",
+    "    )\n",
+    "    .add_params(brush)\n",
+    ")\n",
+    "\n",
+    "bars = (\n",
+    "    alt.Chart(source)\n",
+    "    .mark_bar()\n",
+    "    .encode(y=\"Origin:N\", color=\"Origin:N\", x=\"count(Origin):Q\")\n",
+    "    .transform_filter(brush)\n",
+    ")\n",
+    "\n",
+    "chart = points & bars\n",
+    "\n",
+    "chart"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Log interactive chart to Neptune"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "run[\"altair\"].upload(chart)"
```

Comment:
What do you think about just using `=` where applicable, instead of `upload()`?

Reply:
imo `upload()` makes more sense for file/file-like objects

Reply:
It is "sensible", sure. But I guess we'd ideally use the same logging method (`=`) in as many cases as possible, to avoid getting in the users' way by throwing more function names into the mix. Maybe we should strive to use a specific function name only when some special functionality comes in, like series logging or artifact tracking... Food for thought.

Reply:
Tagged to to a thread internally

Reply:
What I mean is, if the desired action is "log/upload that thing", we should rather do this

```
run["..."] = value
run["..."] = plot
run["..."] = image
run["..."] = anything_really
```

than this (hypothetical example, for the sake of argument)

```
run["..."].log_value(value)
run["..."].upload_plot(plot)
run["..."].upload_image(image)
run["..."].upload_file(anything_really)
```